File: src/Symmetric Tree.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode *root) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(!root)
            return true;
        return complr(root->left, root->right);
    }
    
    bool complr(TreeNode *l, TreeNode *r)
    {
        if(!l && !r)
            return true;
        else if(l && r && l->val == r->val)
        {
            return complr(l->left, r->right) && complr(l->right, r->left);
        }
        else
            return false;
    }
};

File: python/Symmetric Tree.py
#Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param root, a tree node
    # @return a boolean
    def isSymmetric(self, root):
        if root is None:
            return True
        return self.complr(root.left, root.right)

    def complr(self, left, right):
        if left == None and right == None:
            return True

        elif left != None and right != None and left.val == right.val:
            return self.complr(left.left, right.right) and self.complr(left.right, right.left)

        else:
            return False

if __name__ == '__main__':
    so = Solution()
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(2)
    print(so.isSymmetric(root))